modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GMLEncodingUtils.java

```diff
@@ -260,31 +276,39 @@ public XSDTypeDefinition createXmlTypeFromFeatureType(SimpleFeatureType featureT
             if ( attribute instanceof GeometryDescriptor ) {
                 Class binding = attribute.getType().getBinding();
                 if ( Point.class.isAssignableFrom( binding ) ) {
-                    element.setTypeDefinition( schemaIndex.getTypeDefinition(gml.qName("PointPropertyType")));
+                    element.setTypeDefinition(schemaIndex.getTypeDefinition(gml
+                            .qName("PointPropertyType")));
                 }
                 else if ( LineString.class.isAssignableFrom( binding ) ) {
-                    element.setTypeDefinition( schemaIndex.getTypeDefinition(gml.qName("LineStringPropertyType")));
+                    element.setTypeDefinition(schemaIndex.getTypeDefinition(gml
+                            .qName("LineStringPropertyType")));
                 }
                 else if ( Polygon.class.isAssignableFrom( binding) ) {
-                    element.setTypeDefinition( schemaIndex.getTypeDefinition(gml.qName("PolygonPropertyType")));
+                    element.setTypeDefinition(schemaIndex.getTypeDefinition(gml
+                            .qName("PolygonPropertyType")));
                 }
                 else if ( MultiPoint.class.isAssignableFrom( binding ) ) {
-                    element.setTypeDefinition( schemaIndex.getTypeDefinition(gml.qName("MultiPointPropertyType")));
+                    element.setTypeDefinition(schemaIndex.getTypeDefinition(gml
+                            .qName("MultiPointPropertyType")));
                 }
                 else if ( MultiLineString.class.isAssignableFrom( binding ) ) {
-                    element.setTypeDefinition( schemaIndex.getTypeDefinition(gml.qName("MultiLineStringPropertyType")));
+                    element.setTypeDefinition(schemaIndex.getTypeDefinition(gml
+                            .qName("MultiLineStringPropertyType")));
                 }
                 else if ( MultiPolygon.class.isAssignableFrom( binding) ) {
-                    element.setTypeDefinition( schemaIndex.getTypeDefinition(gml.qName("MultiPolygonPropertyType")));
+                    element.setTypeDefinition(schemaIndex.getTypeDefinition(gml
+                            .qName("MultiPolygonPropertyType")));
                 }
                 else {
-                    element.setTypeDefinition( schemaIndex.getTypeDefinition(gml.qName("GeometryPropertyType")));
+                    element.setTypeDefinition(schemaIndex.getTypeDefinition(gml
+                            .qName("GeometryPropertyType")));
```

Comment:
Indeed, curves are supported only starting GML3. And besides that, the way they have been implemented makes them mingle with linear geometries rather seamlessly.